pkg/auth/audit/audit_test.go

```diff
@@ -126,20 +137,178 @@ func Test_concealSensitiveData(t *testing.T) {
 			want:  machineDataWant,
 		},
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+	for i := range tests {
+		test := tests[i]
+		a.Run(test.name, func() {
 			var want map[string]interface{}
-			if err := json.Unmarshal(tt.want, &want); err != nil {
-				t.Errorf("error unmarshaling: %v", err)
-			}
-			got := a.concealSensitiveData(tt.uri, tt.input)
+			err := json.Unmarshal(test.want, &want)
+			a.NoError(err, "failed to unmarshal")
+			got := logger.concealSensitiveData(test.uri, test.input)
 			var gotMap map[string]interface{}
-			if err := json.Unmarshal(got, &gotMap); err != nil {
-				t.Errorf("error unmarshaling: %v", err)
-			}
-			if !reflect.DeepEqual(want, gotMap) {
-				t.Errorf("concealSensitiveData() = %s, want %s", got, tt.want)
+			err = json.Unmarshal(got, &gotMap)
+			a.NoError(err, "failed tto unmarshal")
+			a.Equal(want, gotMap, "concealSensitiveData() = %s, want %s", got, test.want)
+		})
+	}
+}
+func (a *AuditTest) TestCompression() {
+	// Create a temp log file
+	tmpFile, err := os.CreateTemp("", "audit-test")
+	a.Require().NoError(err, "Failed to create temp directory.")
+	// close the file so the logger can open it for writing
+	err = tmpFile.Close()
+	a.Require().NoError(err, "Failed to close temporary file after creation")
+
+	tmpPath := tmpFile.Name()
+	defer func() {
+		err = os.RemoveAll(tmpPath)
+		a.NoError(err, "Failed to clean up temp directory")
+	}()
+
+	writer := NewLogWriter(tmpPath, LevelRequestResponse, 30, 30, 100)
+	a.Require().NotNil(writer, "Failed to create auditWriter.")
+
+	sensitiveRegex, err := regexp.Compile(`[pP]assword|[tT]oken`)
+	a.Require().NoErrorf(err, "Failed to create valid regex: %v", err)
+
+	req, err := http.NewRequest(http.MethodGet, "/test", nil)
+	a.Require().NoErrorf(err, "Failed to create request: %v", err)
+
+	auditLog, err := newAuditLog(writer, req, sensitiveRegex)
+	a.Require().NoErrorf(err, "Failed to create AuditLog: %v", err)
+
+	const testString = "{\"test\":\"response\"}"
+
+	tests := []struct {
+		name         string
+		respHeader   http.Header
+		respBody     []byte
+		returnCode   int
+		expectedBody string
+		Error        error
+	}{
+		{
+			name:       "invalid Encoding",
+			respHeader: http.Header{"Content-Type": []string{"application/json"}, "Content-Encoding": []string{"bzip2"}},
+			respBody:   []byte(testString),
+			Error:      fmt.Errorf(""), // any error
```

Comment:
There would be a specific error in this case, right? Why not go ahead and test for that?

Reply:
The Error didn't have a type since we created it with this line [`err = fmt.Errorf("unknown content-encoding '%s' for response", resHeaders.Get("Content-Encoding"))`](https://github.com/rancher/rancher/blob/c79aae4abcaa6fb8c5ea20007585b03010037cde/pkg/auth/audit/audit.go#L224). Though I can update it to be a exported error and check for that.